game/src/main/scala/io/aigar/game/Resources.scala

```diff
@@ -49,11 +50,32 @@ class Resources(grid: Grid) {
 class ResourceType(grid: Grid, val min: Int, val max: Int, mass: Int, score: Int) {
   var positions = List.fill(max)(grid.randomPosition)
 
-  def spawnResources: Unit = {
+  def spawnResources(players: List[Player]): Unit = {
     val ratio = (positions.length - min).toFloat / (max - min)
 
     if (scala.util.Random.nextFloat >= ratio) {
-      positions :::= List(grid.randomPosition)
+      val loop = new Breaks
+      var inCell = true
+      var position = grid.randomPosition
+      var count =0
+      while (inCell && (count < 10)) {
```

Comment:
I vote that we just try it once. If it fails, we can exit without spawning one. It's not a dramatic event to skip a spawn (we update pretty often) + it would keep the code cleaner imo:
```scala
...
for (player <- players) {
  for (cell <- player.cells) {
    if (cell.contains(position)) return
   }
}
...
```